packages/sdk/src/oracle/store/index.ts

```diff
@@ -0,0 +1,18 @@
+import type { State } from "../types/state";
```

Comment:
I'm a little confused about what in this file is specific to the oracle? This feels like a general module that could be generic and used for _any_ store, no? Is there a reason it isn't/can't be generalized?

Reply:
i took a stab at generalizing this. the problem really is the "Writer" class isnt general and is structured specifically to the state defined in types/state, which describes the shape of the data for the application.  I've made the interface for writing expose this Writer class to the caller which wraps the state to give you a safer way to do mutations.  So thats where the sticking point is for generalizing it currently.

Reply:
i was able to break out the generic store as a separate client, and wrap this with an app specific store